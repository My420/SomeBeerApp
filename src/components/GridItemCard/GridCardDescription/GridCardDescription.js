import React from 'react';
import styles from './GridCardDescription.module.scss';

const GridCardDescription = ({ parameter, title }) => {
  // eslint-disable-next-line no-console
  console.log('render ===== GridCardDescription');

  const description = Object.keys(parameter).map(key => {
    return [
      <dt className={styles.name} key={`${key}-dt`}>
        {key}
      </dt>,
      <dd className={styles.value} key={`${key}-dd`}>
        {parameter[key]}
      </dd>
    ];
  });

  return (
    <div className={styles.description}>
      <p className={styles.title}>{title}</p>
      <dl className={styles.parameter}>{description}</dl>
    </div>
  );
};

export default GridCardDescription;
