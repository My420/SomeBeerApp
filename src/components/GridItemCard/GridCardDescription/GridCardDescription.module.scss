@import '../../../sass/variables.scss';

.description {
  position: absolute;
  top: 0px;
  left: 0px;

  box-sizing: border-box;

  width: 100%;
  height: 100%;

  padding: 3%;

  background-color: $menu-bar-color;
}

.parameter {
  display: flex;
  flex-wrap: wrap;

  width: 40%;

  box-sizing: border-box;

  margin: 10% auto;
}

.name,
.value {
  display: inline-block;
  width: 50%;
  height: auto;

  margin: 0;

  text-transform: uppercase;
}
