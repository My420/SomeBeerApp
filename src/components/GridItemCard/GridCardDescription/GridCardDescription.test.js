import React from 'react';
import { shallow } from 'enzyme';
import GridCardDescription from './GridCardDescription';

describe('test <GridCardDescription/> component', () => {
  const parameter = {
    abv: 11,
    ibu: 22,
    ebc: 33
  };

  const wrapper = shallow(
    <GridCardDescription parameter={parameter} title="test title" />
  );

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
