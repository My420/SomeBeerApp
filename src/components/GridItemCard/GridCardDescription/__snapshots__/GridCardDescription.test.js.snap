// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GridCardDescription/> component should match snapshot 1`] = `
<div
  className="description"
>
  <p
    className="title"
  >
    test title
  </p>
  <dl
    className="parameter"
  >
    <dt
      className="name"
      key="abv-dt"
    >
      abv
    </dt>
    <dd
      className="value"
      key="abv-dd"
    >
      11
    </dd>
    <dt
      className="name"
      key="ibu-dt"
    >
      ibu
    </dt>
    <dd
      className="value"
      key="ibu-dd"
    >
      22
    </dd>
    <dt
      className="name"
      key="ebc-dt"
    >
      ebc
    </dt>
    <dd
      className="value"
      key="ebc-dd"
    >
      33
    </dd>
  </dl>
</div>
`;
